Sources/KlaviyoSwift/Klaviyo.swift

```diff
@@ -259,13 +277,56 @@ public class Klaviyo: NSObject  {
                 , "Property values must be of NSString, NSNumber, NSNull, NSDictionary, NSDate, or NSURL. Got: \(String(describing: properties[k as! NSCopying]))")
         }
     }
+}
+
+
+struct KlaviyoSDK {
+    func initialize(with apiKey: String) {
+        dispatchOnMainThread(action: .initialize(apiKey))
+    }
+    
+    func set(profile: Profile) {
+        dispatchOnMainThread(action: .enqueueProile(profile))
+    }
+    
+    func resetProfile() {
+        dispatchOnMainThread(action: .resetProfile)
+    }
     
-    private func dispatchOnMainThread(action: KlaviyoAction) {
-        Task {
-            await MainActor.run {
-                _ = environment.analytics.store.send(action)
-            }
+    func set(email: String) {
+        dispatchOnMainThread(action: .setEmail(email))
+        
+    }
+    
+    func set(externalId: String) {
+        dispatchOnMainThread(action: .setExternalId(externalId))
+    }
+    
+    func set(profileAttribute: String, value: Any) {
+        // This seems tricky to implement with Any - might need to restrict to something equatable, encodable....
+    }
+    
+    func create(event: Event) {
+        dispatchOnMainThread(action: .enqueueEvent(event))
+    }
+    
+    func set(pushToken: Data) {
+        let apnDeviceToken = pushToken.map { String(format: "%02.2hhx", $0) }.joined()
+        dispatchOnMainThread(action: .setPushToken(apnDeviceToken))
+    }
+    
+    func handle(remoteNotification userInfo: [AnyHashable : Any], fetchCompletionHandler completionHandler: @escaping (UIBackgroundFetchResult) -> Void) {
+        if let properties = userInfo as? [String: Any], let body = properties["body"] as? [String: Any], let _ = body["_k"] {
+            create(event: Event(attributes: .init(metric: .init(name: .OpenedPush), properties: properties, profile: [:])))
+            completionHandler(.noData)
         }
     }
     
+    func handle(notificationResponse: UNNotificationResponse, withCompletionHandler completionHandler: @escaping () -> Void) {
+        if let properties = notificationResponse.notification.request.content.userInfo as? [String: Any],
```

Comment:
should we call the completion handler outside of the `if let` condition else if any one of the if let condition is not satisfied the closure will never get called?

Reply:
I think the idea is we only call the completion handler if it the notification originated from Klaviyo... that said we should probably return a bool so they know whether we called the completion handler

Reply:
Agreed on the bool!

Reply:
Fixed it.